app/src/main/java/com/example/user/my_calculator/MainActivity.java

```diff
@@ -12,290 +12,132 @@
-            }
-        });
+    public void onCClick(View view) {
+        firstNumber = NaN;
+        txtResult.setText(null);
+        edtInput.setText(null);
+    }
 
-        btnNumber3.setOnClickListener(new View.OnClickListener() {
-            @SuppressLint("SetTextI18n")
-            @Override
-            public void onClick(View v) {
-                edtInput.setText(edtInput.getText() + "3");
-                edtInput.setSelection(edtInput.getText().length());
-            }
-        });
+    public void onDelClick(View view) {
+        String number = edtInput.getText().toString();
+        if (number.length() > 0) {
+            number = number.substring(0, number.length() - 1);
+        }
+        edtInput.setText(number);
+        edtInput.setSelection(edtInput.getText().length());
+    }
 
-        btnNumber4.setOnClickListener(new View.OnClickListener() {
-            @SuppressLint("SetTextI18n")
-            @Override
-            public void onClick(View v) {
-                edtInput.setText(edtInput.getText() + "4");
-                edtInput.setSelection(edtInput.getText().length());
-            }
-        });
+    @SuppressLint("SetTextI18n")
+    public void onDigitClick(View view) {
+        Button digit = (Button) view;
 
-        btnNumber5.setOnClickListener(new View.OnClickListener() {
-            @SuppressLint("SetTextI18n")
-            @Override
-            public void onClick(View v) {
-                edtInput.setText(edtInput.getText() + "5");
-                edtInput.setSelection(edtInput.getText().length());
-            }
-        });
+        edtInput.setText(edtInput.getText() + digit.getText().toString());
+        edtInput.setSelection(edtInput.getText().length());
+    }
 
-        btnNumber6.setOnClickListener(new View.OnClickListener() {
-            @SuppressLint("SetTextI18n")
-            @Override
-            public void onClick(View v) {
-                edtInput.setText(edtInput.getText() + "6");
-                edtInput.setSelection(edtInput.getText().length());
-            }
-        });
+    @SuppressLint("SetTextI18n")
+    public void onAddClick(View view) {
+        action = "+";
+        if (!Double.isNaN(firstNumber)) {
+            secondNumber = Double.parseDouble(edtInput.getText().toString());
+            firstNumber = firstNumber + secondNumber;
+        } else {
+            firstNumber = Double.parseDouble(edtInput.getText().toString());
+        }
+
+        txtResult.setText(firstNumber + " + ");
+        edtInput.setText(null);
+    }
 
-        btnNumber7.setOnClickListener(new View.OnClickListener() {
-            @SuppressLint("SetTextI18n")
-            @Override
-            public void onClick(View v) {
-                edtInput.setText(edtInput.getText() + "7");
-                edtInput.setSelection(edtInput.getText().length());
-            }
-        });
+    @SuppressLint("SetTextI18n")
+    public void onSubClick(View view) {
```

Comment:
onSubClick/onAddClick/onMulClick/onDivClick the same methods! one thing that differs is operation. try to rework them and leave only one method which has common logic and move all operation's dependant part to enum/class etc and use polymorphic behavior.